libsolidity/ast/AST.h

```diff
@@ -1374,20 +1374,33 @@ class TryStatement: public Statement
 		ASTPointer<ASTString> const& _docString,
 		ASTPointer<Expression> const& _externalCall,
 		std::vector<ASTPointer<TryCatchClause>> const& _clauses
-	):
-		Statement(_id, _location, _docString),
-		m_externalCall(_externalCall),
-		m_clauses(_clauses)
-	{}
+	);
+
 	void accept(ASTVisitor& _visitor) override;
 	void accept(ASTConstVisitor& _visitor) const override;
 
 	Expression const& externalCall() const { return *m_externalCall; }
-	std::vector<ASTPointer<TryCatchClause>> const& clauses() const { return m_clauses; }
+
+	ASTPointer<TryCatchClause> successClause() const noexcept { return m_success; }
+	ASTPointer<TryCatchClause> structuredClause() const noexcept { return m_structured; }
+	ASTPointer<TryCatchClause> fallbackClause() const noexcept { return m_fallback; }
```

Comment:
As mentioned in chat: the vector of clauses has to be stored as is in the constructor, so that we can properly type-check it in the TypeChecker.
These helpers will then have to search through the stored vector.